File: builder-pattern/python/src/ManualBuilder.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.Builder import Builder
from src.Manual import Manual

class ManualBuilder(Builder):
  manual = None

  def __init__(self):
    pass

  def reset(self):
    self.manual = Manual()

  def setName(self, name):
    self.manual.setName(name)
    return self

  def setScreen(self, screen):
    self.manual.setScreen(screen)
    return self

  def setGPU(self, no):
    self.manual.setGpuType(no)
    return self

  def getProduct(self):
    return self.manual

File: builder-pattern/go/src/ManualBuilder.go
package src

type ManualBuilder struct {
  manual *Manual
}

func (m *ManualBuilder) Reset() {
  m.manual = new(Manual)
}

func (m *ManualBuilder) SetName(name string) Builder {
  m.manual.SetName(name)
  return (Builder)(m)
}

func (m *ManualBuilder) SetScreen(screen []int) Builder {
  m.manual.SetScreen(screen)
  return (Builder)(m)
}

func (m *ManualBuilder) SetGPU(no int) Builder {
  m.manual.SetGpuType(no)
  return (Builder)(m)
}

func (m *ManualBuilder) GetProduct() *Manual {
  return m.manual
}
